src/LodeRunner.API/src/Utilities/Parameters/ParametersValidator.cs

```diff
@@ -14,48 +15,48 @@ namespace LodeRunner.API.Middleware
     public class ParametersValidator<TEntity>
         where TEntity : class
     {
-        /// <summary>
-        /// Validate Id.
-        /// </summary>
-        /// <param name="entityId">id to validate.</param>
-        /// <returns>empty list on valid.</returns>
-        public static List<ValidationError> ValidateEntityId(string entityId)
-        {
-            List<ValidationError> errors = new ();
-
-            string entityIdFieldName = $"{typeof(TEntity).Name}Id";
-
-            if (!IsEntityIdValid(entityId))
-            {
-                errors.Add(new ValidationError() { Target = entityIdFieldName, Message = ValidationError.GetErrorMessage(entityIdFieldName) });
-            }
-
-            return errors;
-        }
-
-        /// <summary>
-        /// Check if Entity ID is Valid.
-        /// </summary>
-        /// <param name="entityId">id to validate.</param>
-        /// <returns>true on valid.</returns>
-        private static bool IsEntityIdValid(string entityId)
-        {
-            if (string.IsNullOrWhiteSpace(entityId))
-            {
-                return false;
-            }
-
-            Guid guidValue;
-            try
-            {
-                guidValue = Guid.Parse(entityId);
-            }
-            catch (Exception)
-            {
-                return false;
-            }
-
-            return true;
-        }
+        /// <summary>
+        /// Validate Id.
+        /// </summary>
+        /// <param name="entityId">id to validate.</param>
+        /// <returns>List of error messages.</returns>
+        public static List<string> ValidateEntityId(string entityId)
+        {
+            List<string> errors = new ();
+
+            string entityIdFieldName = $"{typeof(TEntity).Name}Id";
+
+            if (!IsValid(entityId))
+            {
+                errors.Add($"{entityIdFieldName} - {ValidationError.GetErrorMessage(entityIdFieldName)}");
+            }
+
+            return errors;
+        }
+
+        /// <summary>
+        /// Check if Entity ID is Valid.
+        /// </summary>
+        /// <param name="entityId">id to validate.</param>
+        /// <returns>true on valid.</returns>
+        private static bool IsValid(string entityId)

```

Comment:
merge this private method into ValidateEntityId(string entityId) or at least rename into IsValidGuid?

Reply:
Great comment. Merged into ValidateEntityId.